app/models/profile.rb

```diff
@@ -89,20 +89,27 @@ def major_os_version
   alias os_major_version major_os_version
 
   def update_rules(ids: nil, ref_ids: nil)
```

Comment:
This method wasn't quite right - it wasn't able to add a rule and delete a rule at the same time. There's a difference between rule_ids_to_add (just the additional rules) and new_rules (the complete set of rules the profile should have after this update). We want to delete `rule_ids - new_rules`, not `rule_ids - rule_ids_to_add`.